zebra-state/src/service/finalized_state/disk_format/upgrade.rs

```diff
@@ -74,6 +78,31 @@ pub struct DbFormatChangeThreadHandle {
     should_cancel_format_change: Arc<AtomicBool>,
 }
 
+/// Defines method signature for running disk format upgrades.
+pub trait DiskFormatUpgrade {
+    /// Runs disk format upgrade.
+    fn run(
+        &self,
+        initial_tip_height: Height,
+        network: Network,
+        upgrade_db: &ZebraDb,
+        should_cancel_format_change: Arc<AtomicBool>,
+    );
+}
+
+fn format_upgrades() -> Vec<(Version, Box<dyn DiskFormatUpgrade>)> {
+    fn parse_expect_version(version: &str) -> Version {
+        version
+            .parse()
+            .expect("Hardcoded version string should be valid.")
+    }
+
+    vec![
+        (parse_expect_version("25.1.0"), Box::new(PruneTrees)),
+        (parse_expect_version("25.2.0"), Box::new(AddSubtrees)),
+    ]
+}
```

Comment:
This is a good refactor, but large code movements should go in a separate PR, without any changes to functionality.

That makes them much easier to review and test.

Reply:
Actually, I think it might be better to delay this refactor for a while, because it will cause merge conflicts with bug fixes to the `25.1` state upgrade.

Reply:
> Actually, I think it might be better to delay this refactor for a while, because it will cause merge conflicts with bug fixes to the 25.1 state upgrade.

I think it could help with avoiding merge conflicts if we do the upgrade for `25.2.1` in parallel to `25.1.1`.

Reply:
> > Actually, I think it might be better to delay this refactor for a while, because it will cause merge conflicts with bug fixes to the 25.1 state upgrade.
> 
> I think it could help with avoiding merge conflicts if we do the upgrade for `25.2.1` in parallel to `25.1.1`.

I'm not sure what you mean by "in parallel"". Do you mean "concurrently running in a separate thread", "in the same PR", or "in the same release"?

But it's not just merge conflicts that are a factor here, it's technical complexity risk, reviewer attention, testing, and developer time/scope. Reducing code changes helps with all of these.